creusot/src/backend/program.rs

```diff
@@ -1081,23 +1087,29 @@ impl<'tcx> Statement<'tcx> {
                 istmts.extend(assign);
                 istmts
             }
-            Statement::Resolve { did, subst, pl, pat } => {
+            Statement::Resolve { did, subst, pl } => {
                 lower.ctx.translate(did);
                 let mut istmts = Vec::new();
 
                 let rp = Exp::qvar(lower.names.value(did, subst));
                 let loc = pl.local;
 
-                let mut exp = rp.app_to(pl.as_rplace(lower, &mut istmts));
-                if let Some(pat) = pat {
-                    exp = Exp::Match(
+                let bound = lower.fresh_sym_from("inv_asrt");
```

Comment:
"inv_asrt" does not make any sense as an identifier. Even somthing like "x" would be better.